backend/src/User/user.service.ts

```diff
@@ -37,6 +39,59 @@ class Service {
         };
     }
 
+    public async sendForgotPassword(email: string) {
+        const user = await UserModel.findByEmailAuth(email);
+        if (!user) {
+            return { status: false, error: 'Email does not match any account.' };
+        }
+
+        // Create token that expires in 1 hour
+        const token = crypto.randomBytes(20).toString('hex');
+        user.resetPasswordToken = token;
+        user.resetPasswordExpires = Date.now() + 3600000;
+
+        UserModel.updateById(user.userId, user);
+
+        const success = sendEmail.sendPasswordRecoveryEmail(user.email, token);
+        if (!success) {
+            return {
+                status: false,
+                error:
+                    'Error in sending email to ' +
+                    user.email +
+                    '. Please contact your administrator'
+            };
+        }
+        return {
+            status: true,
+            message: 'An email has been sent to ' + user.email + ' with further instructions.'
+        };
+    }
+
+    public async resetPassword(token?: string, password?: string) {
+        if (!token || !password) {
+            return { status: false, error: 'Invalid authorization header' };
+        }
+        const user = await UserModel.findByResetPasswordToken(token);
+
+        if (!user) {
+            return {
+                status: false,
+                error: 'Reset password token is invalid or has expired.'
+            };
+        }
+        user.password = bcrypt.hashSync(password, 10);
```

Comment:
The salt should not be hard coded in this case

Reply:
env variables?

Reply:
now is coded as 10

Reply:
we need to always use the same salt because the hashing would not work if we keep the same db with old salt and we restart the server with a new one